frontend/src/components/Collections/components/Grid/components/CollectionsGrid/index.tsx

```diff
@@ -42,29 +52,44 @@ const CollectionsGrid: FC<Props> = ({
 function renderCollections(
   collections: CollectionResponse[],
   displayVisibility?: VISIBILITY_TYPE,
-  accessType?: ACCESS_TYPE
+  accessType?: ACCESS_TYPE,
+  revisionsEnabled?: boolean
 ) {
   const collectionElements = [] as Array<ReactChild>;
-  sortByCreatedAtDescending(collections).forEach(({ id, visibility }) => {
-    if (!displayVisibility || visibility === displayVisibility) {
-      collectionElements.push(
-        <CollectionRow
-          id={id}
-          key={id + visibility}
-          visibility={visibility}
-          accessType={accessType}
-        />
-      );
+  const revisions = revisionsEnabled ? findRevisions(collections) : false;
+  sortByCreatedAtDescending(revisions || collections).forEach(
+    ({ id, visibility, revision }) => {
```

Comment:
revision can be true/false/undefined

Reply:
true: means there is a revision
false: means there could but isn't
undefined: we're on 

- the my collections page, no revision logic